src/pudl/extract/csv.py

```diff
@@ -0,0 +1,153 @@
+"""Extractor for CSV data."""
+from csv import DictReader
+from functools import lru_cache
+from importlib import resources
+from pathlib import Path
+from zipfile import ZipFile
+
+import pandas as pd
+import sqlalchemy as sa
+
+from pudl import logging_helpers
+from pudl.workspace.datastore import Datastore
+
+logger = logging_helpers.get_logger(__name__)
+
+
+class CsvReader:
```

Comment:
This is a simple first iteration of `FercDbfReader`, but for CSVs, attempting to reuse and establish consistent patterns. Some of the methods are copy-pasted in this draft and should be extracted to common functionality or otherwise reconciled to ward against complexity and drift.

I also could have used the objects in the `excel` module as the basis, which have their own patterns. But they appear to handle more complexity than I'm seeing at this point for CSVs. The overlap between the functionality needed for CSVs and DBFs appears stronger.

Reply:
So the `CsvReader` provides:

* a list of tables that we know how to extract from an archive
* a `CsvArchive` object that we can get actual table data from

Is that right?

If so, what do you think about merging the `Archive` and the `Reader` into one class that handles "get a table from an archive that contains CSV files?"

Reply:
Yeah I was thinking similar. Will do!